src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -112,12 +128,56 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         )
         check_cloud_sync = asyncio.ensure_future(rclone_check_progress(job, proc))
         await proc.wait()
+        await asyncio.wait_for(check_cloud_sync, None)
+
+        if snapshot:
+            await middleware.call("zfs.snapshot.remove", snapshot)
+
         if proc.returncode != 0:
-            await asyncio.wait_for(check_cloud_sync, None)
             raise ValueError("rclone failed")
+
+        await run_script(job, cloud_sync["post_script"], "Post-script")
+
         return True
 
 
+async def run_script(job, hook, name):
+    hook = hook.strip()
+    if not hook:
+        return
+
+    if not hook.startswith("#!"):
+        hook = f"#!/bin/bash\n{hook}"
+
+    fd, name = tempfile.mkstemp()
+    os.close(fd)
+    try:
+        os.chmod(name, 0o700)
+        with open(name, "w+") as f:
+            f.write(hook)
+
+        proc = await Popen(
+            [name],
```

Comment:
Just executing the script sounds useless, wont you agree?

We need to pass some useful args to it, e.g. path, dataset, cloud task id? What else?

Reply:
I've thought that script knows what to do itself. It still would be great to pass some parameters. There are so many of them and I can't think of anything better than environment variables.

Reply:
It cant know what to do if it doesnt know what is the task being execute, nor the path.

Environment variables sounds good to me.

Reply:
https://redmine.ixsystems.com/issues/29908?issue_count=27&issue_position=8&next_issue_id=28024&prev_issue_id=32460#note-8